File: curdproject/src/main/java/com/professional/curdproject/models/Employee.java
package com.professional.curdproject.models;

import java.math.BigDecimal;
import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@ToString
public class Employee {
@Id
@GeneratedValue(strategy=GenerationType.AUTO)
@Column(name="Employee_Id",nullable=false)
 private long empId;
@Column(name="Name",nullable=false,length=50)
private String empName;
@Column(name="Salary",nullable=false)
private BigDecimal empSalary;
@Embedded
private EmployeeAddress employeeAddress;
@ManyToOne
@JoinColumn(name="Department")
private Department department;

}

File: employee-data/src/app/common/employee.ts
import { Department } from './department'


export class Employee {
    
    empName:string
    empSalary:number 
    department:Department
    employeeAddress :{
        city:string ,
        street:string,
        houseNum:number
    }
    constructor(
        empName:string,
        empSalary:number,
        department:Department,
        employeeAddress :{
            city:string ,
            street:string,
            houseNum:number
        }
        )
        {        
        this.empName=empName
        this.empSalary=empSalary
        this.department=department
        this.employeeAddress=employeeAddress
    }
}
